src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -280,6 +284,41 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
             return result.ToImmutable();
         }
 
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: MethodDeclarationSyntax method } })
+            {
+                return;
+            }
+
+            if (method.Parent is null)
+                return;
+
+            var methodParameterType = context.SemanticModel.GetTypeInfo(parameterType, cancellationToken).Type;
+            if (methodParameterType is null)
+                return;
+
+            var siblingMethods = method.Parent.ChildNodes().OfType<MethodDeclarationSyntax>().Where(m => m != method && m.Identifier.Text == method.Identifier.Text);
```

Comment:
Wouldn't it make more sense to look for overloads in the compilation/semantic model? That way you could also take overloads from base classes or parameter names from interface declarations into account (maybe names from implicit interface implementations are already suggested. Haven't checked that).

Reply:
Sounds reasonable.
@sharwell @CyrusNajmabadi Which approach do you prefer?

Reply:
We can use the semantic model as long as we use the one with frozen partial semantics (completion performance depends on this). I would be comfortable filing this as a follow-up issue.